.github/workflows/lint-golangci.yml

```diff
@@ -12,7 +12,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v4
         with:
-          go-version: '1.21'
+          go-version-file: 'go.mod'
```

Comment:
To reviewer:
I unified the way to get go version from go.mod, check the detail here https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file